packages/crypto/src/KeySignature.ts

```diff
@@ -0,0 +1,25 @@
+import Address, { KeyTypes } from '@helium/address'
+
+export default class KeySignature {
+  public index!: number
+
+  public signature!: Uint8Array
+
+  constructor(index: number, signature: Uint8Array) {
+    this.index = index
+    this.signature = signature
+  }
+
+  public static new(addresses: Address[], address: Address, signature: Uint8Array): KeySignature {
```

Comment:
could this be handled in the constructor itself?

Reply:
I'd personally leave them separate, I think this pattern allows `new` to be used in cases where you want to use the list of addresses to find the signature, but you may have an aggregator accepting serialized `KeySignature` objects (from `KeySignature(...).bin`) instead, in which case the constructor would be more appropriate to use.

Reply:
My concern with having them be separate is that it appears that there is essential validation happening in this static method for what comprises a valid instance of a KeySignature. It's idiomatic javascript to use the `new` keyword to instantiate a class. so my worry is that we're opening up the possibility of creating invalid instances if a user reaches for a `new KeySingature` before they see the `KeySignature.create` method.

Reply:
I'm less familiar with idiomatic javascript, but that makes sense to me 👍  I'll update the method signatures and move things around. @allenan - Would you suggest eliminating the `create` method altogether?

Reply:
if the only purpose of the create method is to instantiate a valid instance of the class, then yes, that's for the constructor to do. if it has another purpose, then maybe we can look at another name for it?